types/time.go

```diff
@@ -823,12 +823,16 @@ func parseDatetime(sc *stmtctx.StatementContext, str string, fsp int8, isFloat b
 
 	tmp := FromDate(year, month, day, hour, minute, second, microsecond)
 	if overflow {
-		// Convert to Go time and add 1 second, to handle input like 2017-01-05 08:40:59.575601
-		t1, err := tmp.GoTime(gotime.Local)
-		if err != nil {
+		if sc != nil {
+			// Convert to Go time and add 1 second, to handle input like 2017-01-05 08:40:59.575601
+			t1, err := tmp.GoTime(sc.TimeZone)
```

Comment:
maybe need a test case to reproduce the bug~? I try replace this line back to `time.Local` and all test case still passed.

Reply:
Added.

Reply:
You can take a look, if my test is correct I will add more test cases.

Reply:
soory, still hava problem.

Reply:
I think it's difficult to add a test case to reproduce the bug. Could you please give me some tips?

Reply:
Seems that there is no bug in the previous code since the overflow logic is not related to the time zone, then using `gotime.Local` should be fine...I'm not 100% sure about this.

However, I still believe it's better to remove `gotime.Local`.

Reply:
Yes, I think so. I will remove the useless test.